evm/bridge-token-factory/contracts/BridgeTokenFactory.sol

```diff
@@ -241,33 +169,118 @@ contract BridgeTokenFactory is
         );
     }
 
-    function finTransferExtension(FinTransferPayload memory payload) internal virtual {}
+    function mintTokenExtension(BridgeTypes.MintTokenPayload memory payload) internal virtual {}
 
-    function initTransfer(
+    function burnToken(
         string calldata token,
         uint128 amount,
         uint128 fee,
         uint128 nativeFee,
         string calldata recipient
-    ) payable external whenNotPaused(PAUSED_INIT_TRANSFER) {
-        initTransferNonce += 1;
-        _checkWhitelistedToken(token, msg.sender);
-        require(_isBridgeToken[_nearToEthToken[token]], "ERR_NOT_BRIDGE_TOKEN");
+    ) payable external whenNotPaused(PAUSED_BURN_TOKEN) {
+        currentNonce += 1;
+        require(isBridgeToken[nearToEthToken[token]], "ERR_NOT_BRIDGE_TOKEN");
         if (fee >= amount) {
             revert InvalidFee();
         }
 
-        address tokenAddress = _nearToEthToken[token];
+        address tokenAddress = nearToEthToken[token];
 
         BridgeToken(tokenAddress).burn(msg.sender, amount);
 
         uint256 extensionValue = msg.value - nativeFee;
-        initTransferExtension(initTransferNonce, token, amount, fee, nativeFee, recipient, msg.sender, extensionValue);
+        burnTokenExtension(currentNonce, token, amount, fee, nativeFee, recipient, msg.sender, extensionValue);
+
+        emit BridgeTypes.BurnToken(msg.sender, tokenAddress, currentNonce, token , amount, fee, nativeFee, recipient);
+    }
+
+    function burnTokenExtension(
+        uint128 nonce,
+        string calldata token,
+        uint128 amount,
+        uint128 fee,
+        uint128 nativeFee,
+        string calldata recipient,
+        address sender,
+        uint256 value
+    ) internal virtual {}
+
+    function lockToken(
+        address tokenAddress,
+        uint128 amount,
+        uint128 fee,
+        uint128 nativeFee,
+        string calldata recipient,
+        string calldata message
```

Comment:
What is the message and why do we have it here but not for `burn`

Reply:
This one should also be added for `burnToken`, to replace the `account_id:msg` format

Reply:
Then I guess is a question whether we want to separate lock/burn and unlock/mint functionality or have all under init and finalize. The interfaces look to be the same so I think I prefer keeping initializeTransfer and finalizeTransfer.

Reply:
Yeh, we can try to unify it, but we need to consider that the bind token isn't obligatory for now, so we should be careful what kind of token address we pass.